modules/ossm-configuring-distr-tracing-tempo.adoc

```diff
@@ -5,20 +5,66 @@ This module is included in the following assemblies:
 
 :_mod-docs-content-type: PROCEDURE
 [id="ossm-configuring-distr-tracing-tempo_{context}"]
-= Configuring the distributed tracing platform (Tempo)
+= Configuring the distributed tracing platform (Tempo) and the Red Hat build of OpenTelemetry
 
-You can expose tracing data to the {TempoName} stack by appending a named element and the `zipkin` provider to the `spec.meshConfig.extensionProviders` specification in the `ServiceMehControlPlane`, as shown in the following example. Then, a telemetry custom resource configures Istio proxies to collect trace spans and send them to the Tempo distributor service endpoint.
+You can expose tracing data to the {TempoName} by appending a named element and the `opentelemetry` provider to the `spec.meshConfig.extensionProviders` specification in the `ServiceMeshControlPlane`. Then, a telemetry custom resource configures Istio proxies to collect trace spans and send them to the OpenTelemetry Collector endpoint.
 
-You can create a TempoStack instance in the `tracing-system` namespace _after_ creating the `ServiceMeshControlPlane` and the `ServiceMeshMemberRoll` resources.
+//There is no common attribute for OpenTelemetry Collector.
+
+You can create a {OTELName} instance in a mesh namespace and configure it to send tracing data to a tracing platform backend service.
 
 .Prerequisites
 
-* You have installed the {TempoOperator} and {SMProductName} Operator in the `openshift-operators` namespace.
-* You have created namespaces such as `istio-system` and `tracing-system`.
+* You have created a TempoStack instance using the Red Hat {TempoOperator} in the `tracing-system` namespace. For more information, see "Installing the distributed tracing platform (Tempo)" in the "Additional resources" section.
+
+* You have installed the {OTELOperator} in the namespace `openshift-operators`. For more information, see "Installing the {OTELName}" in the "Additional resources" section.
 
 .Procedure
 
-. Configure the `ServiceMeshControlPlane` resource to define an extension provider:
+. Create an OpenTelemetry Collector instance in a mesh `namespace` such as `bookinfo`:
++
+[source, yaml]
+----
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: otel
+  namespace: bookinfo # <1>
+spec:
+  mode: deployment
+  config: |
+    receivers:
+      jaeger:
+        protocols:
+          grpc:
+          thrift_binary:
+          thrift_compact:
+          thrift_http:
+      otlp:
+        protocols:
+          grpc:
+            endpoint: 0.0.0.0:4317
+          http:
+      zipkin:
+    processors:
+      batch:
```

Comment:
```suggestion
      jaeger:
        protocols:
          grpc: {}
          thrift_binary: {}
          thrift_compact: {}
          thrift_http: {}
      otlp:
        protocols:
          grpc:
            endpoint: 0.0.0.0:4317
          http: {}
      zipkin: {}
    processors:
      batch: {}
```

Reply:
or we can completely remove it from there if they don't need to be filled in

Reply:
> or we can completely remove it from there if they don't need to be filled in

Yes, I think this would be preferable if explicitly setting the values to empty objects is not required.

Reply:
It follows the sample from the Distributed Tracing instructions https://docs.openshift.com/container-platform/4.15/observability/otel/otel-installing.html 

Will Service Mesh users be confused if we remove them?

Reply:
on that page, there is `customize the OpenTelemetryCollector custom resource (CR) with the OTLP, Jaeger, Zipkin receivers and the debug exporter.` 

on our page, we have prefilled `.otlp.protocols.grpc.endpoint: 0.0.0.0:4317 `. If someone is experienced with OpenTelemetryCollector in order to set different values, they will probably know other values that can be filled as well and we don't need to prefill them. ( just my opinion )

Reply:
or we can add there something like this: 




Create an OpenTelemetry Collector instance in a mesh namespace such as bookinfo:
```
apiVersion: opentelemetry.io/v1alpha1
kind: OpenTelemetryCollector
metadata:
  name: otel
  namespace: bookinfo 
spec:
  mode: deployment
  config: |
    receivers:
      otlp:
        protocols:
          grpc:
            endpoint: 0.0.0.0:4317
    exporters:
      otlp:
        endpoint: "tempo-sample-distributor.tracing-system.svc.cluster.local:4317"
        tls:
          insecure: true
    service:
      pipelines:
        traces:
          receivers: [jaeger, otlp, zipkin]
          processors: [batch]
          exporters: [otlp]
```
```
NOTE: You can customize the OpenTelemetryCollector custom resource (CR) with others values as well as is described in  Red Hat build of OpenTelemetry documentation
```

Reply:
+1 for this (assuming it does not introduce any validation errors).

Reply:
Have we verified that there are no validation errors with this suggested change?

Reply:
yes, only the second warning left, 
```
Warning: OpenTelemetryCollector v1alpha1 is deprecated. Migrate to v1beta1.
opentelemetrycollector.opentelemetry.io/otel created
```
however, there is an error in the otel pod
```Error: invalid configuration: service::pipelines::traces: references receiver "jaeger" which is not configured
2024/06/26 07:25:25 collector server run finished with error: invalid configuration: service::pipelines::traces: references receiver "jaeger" which is not configured
```
because `jaeger` and `zipkin` are defined in the config `service.pipelines.traces.receivers`  (The interesting thing is that when you add an empty object there, it is ok )

so, the yaml should be without `zipkin` and `jaeger` in the receivers and `batch` in the processor too  (they are not defined in the yaml anyway so I would say no need to specified that, I don't see the reason why):
```
apiVersion: opentelemetry.io/v1alpha1
kind: OpenTelemetryCollector
metadata:
  name: otel
  namespace: bookinfo 
spec:
  mode: deployment
  config: |
    receivers:
      otlp:
        protocols:
          grpc:
            endpoint: 0.0.0.0:4317
    exporters:
      otlp:
        endpoint: "tempo-sample-distributor.tracing-system.svc.cluster.local:4317"
        tls:
          insecure: true
    service:
      pipelines:
        traces:
          receivers: [otlp]
          exporters: [otlp]
```

I have tried it with the config above ^ and it works. But better wait for @yxun to confirm.

Reply:
Thanks, Matej. I have added a suggested change using your simplified yaml example.

Reply:
Hi @Matej, I tried your simplified example above . However, it's exporter failed to resolve the Tempo distributor service. 

I will add another suggested change example later.

Reply:
@eoinfennessy @mkralik3 How did you get those "spec.config has null objects:" logs ?
I tried my initial example yaml and here is my log below

```
2024-07-01T17:42:28.870Z	info	service@v0.102.1/service.go:113	Setting up own telemetry...
2024-07-01T17:42:28.870Z	info	service@v0.102.1/telemetry.go:96	Serving metrics	{"address": ":8888", "level": "Normal"}
2024-07-01T17:42:28.870Z	warn	jaegerreceiver@v0.102.0/factory.go:49	jaeger receiver will deprecate Thrift-gen and replace it with Proto-gen to be compatbible to jaeger 1.42.0 and higher. See https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/18485 for more details.	{"kind": "receiver", "name": "jaeger", "data_type": "traces"}
2024-07-01T17:42:28.883Z	info	service@v0.102.1/service.go:180	Starting otelcol...	{"Version": "0.102.1", "NumCPU": 4}
2024-07-01T17:42:28.883Z	info	extensions/extensions.go:34	Starting extensions...
2024-07-01T17:42:28.883Z	warn	internal@v0.102.1/warning.go:42	Using the 0.0.0.0 address exposes this server to every network interface, which may facilitate Denial of Service attacks. Enable the feature gate to change the default and remove this warning.	{"kind": "receiver", "name": "jaeger", "data_type": "traces", "documentation": "https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md#safeguards-against-denial-of-service-attacks", "feature gate ID": "component.UseLocalHostAsDefaultHost"}
2024-07-01T17:42:28.883Z	warn	internal@v0.102.1/warning.go:42	Using the 0.0.0.0 address exposes this server to every network interface, which may facilitate Denial of Service attacks. Enable the feature gate to change the default and remove this warning.	{"kind": "receiver", "name": "jaeger", "data_type": "traces", "documentation": "https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md#safeguards-against-denial-of-service-attacks", "feature gate ID": "component.UseLocalHostAsDefaultHost"}
2024-07-01T17:42:28.883Z	warn	internal@v0.102.1/warning.go:42	Using the 0.0.0.0 address exposes this server to every network interface, which may facilitate Denial of Service attacks. Enable the feature gate to change the default and remove this warning.	{"kind": "receiver", "name": "otlp", "data_type": "traces", "documentation": "https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md#safeguards-against-denial-of-service-attacks", "feature gate ID": "component.UseLocalHostAsDefaultHost"}
2024-07-01T17:42:28.883Z	info	otlpreceiver@v0.102.1/otlp.go:102	Starting GRPC server	{"kind": "receiver", "name": "otlp", "data_type": "traces", "endpoint": "0.0.0.0:4317"}
2024-07-01T17:42:28.883Z	warn	internal@v0.102.1/warning.go:42	Using the 0.0.0.0 address exposes this server to every network interface, which may facilitate Denial of Service attacks. Enable the feature gate to change the default and remove this warning.	{"kind": "receiver", "name": "otlp", "data_type": "traces", "documentation": "https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md#safeguards-against-denial-of-service-attacks", "feature gate ID": "component.UseLocalHostAsDefaultHost"}
2024-07-01T17:42:28.883Z	info	otlpreceiver@v0.102.1/otlp.go:152	Starting HTTP server	{"kind": "receiver", "name": "otlp", "data_type": "traces", "endpoint": "0.0.0.0:4318"}
2024-07-01T17:42:28.883Z	warn	internal@v0.102.1/warning.go:42	Using the 0.0.0.0 address exposes this server to every network interface, which may facilitate Denial of Service attacks. Enable the feature gate to change the default and remove this warning.	{"kind": "receiver", "name": "zipkin", "data_type": "traces", "documentation": "https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md#safeguards-against-denial-of-service-attacks", "feature gate ID": "component.UseLocalHostAsDefaultHost"}
2024-07-01T17:42:28.883Z	info	service@v0.102.1/service.go:206	Everything is ready. Begin running and processing data.
2024-07-01T17:42:28.883Z	warn	localhostgate/featuregate.go:63	The default endpoints for all servers in components will change to use localhost instead of 0.0.0.0 in a future version. Use the feature gate to preview the new default.	{"feature gate ID": "component.UseLocalHostAsDefaultHost"}
```